### PR TITLE
Create Exploration_SampleEntropy_IMU.py

### PR DESCRIPTION
**Pull request by kharrison89**
Specifically, want to double check the rotation of the acceleration time series. Its based on the validation trail imu code, so hopefully is good but want to be sure.

### CODE REVIEW

File: Exploration_SampleEntropy_IMU.py [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+    
+    for count,jj in enumerate(GS[:-1]):
+        # Obtain the acceleration and gyroscope from foot contact (HS) to the
+        # subsiquent midstance (MS). MS is needed for linear drift correction
+        
+        #count = 0
+        #jj = GS[count]
+        
+        jjn = GS[count +1]
+        acc_stride_plus = timeseries[HS[jj]:MS[jjn],1:4]
```
Comment (EricHonertBOA) [inline Exploration_SampleEntropy_IMU.py:245]: I think that you have the accelerometer and gyroscope signals flipped here in lines 245 and 246. On lines 376, 383, and 390 the gyro is concatenated prior to the accelerometer

File: Exploration_SampleEntropy_IMU.py [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+           theta = theta-(thetai+thetai_up)/2
+           thetai_up = thetai
+        
+        # Convert to radians
+        theta = theta*np.pi/180
+        # Rotate the accelerometer into the lab coordinate system (LCS)
+        
+        # Note: this rotation has been converted to list comprehension for speed purposes
+        acc_stride_plus_LCS = [(np.array([[cos(ang[2]),-sin(ang[2]),0],[sin(ang[2]),cos(ang[2]),0],[0,0,1]])@np.array([[1,0,0],[0,cos(ang[0]),-sin(ang[0])],[0,sin(ang[0]),cos(ang[0])]])@np.array([[cos(ang[1]),0,sin(ang[1])],[0,1,0],[-np.sin(ang[1]),0,cos(ang[1])]])@acc_stride_plus[kk,:].T).T-np.array([9.81,0,0]) for kk,ang in enumerate(theta)]
+        acc_stride_plus_LCS = np.stack(acc_stride_plus_LCS)
```
Comment (EricHonertBOA) [inline Exploration_SampleEntropy_IMU.py:280]: Are you trying to transpose the matrix here? I would use .T (similar to line 264) to be consistent with the code

File: Exploration_SampleEntropy_IMU.py [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+        else: 
+           theta = theta-(thetai+thetai_up)/2
+           thetai_up = thetai
+        
+        # Convert to radians
+        theta = theta*np.pi/180
+        # Rotate the accelerometer into the lab coordinate system (LCS)
+        
+        # Note: this rotation has been converted to list comprehension for speed purposes
+        acc_stride_plus_LCS = [(np.array([[cos(ang[2]),-sin(ang[2]),0],[sin(ang[2]),cos(ang[2]),0],[0,0,1]])@np.array([[1,0,0],[0,cos(ang[0]),-sin(ang[0])],[0,sin(ang[0]),cos(ang[0])]])@np.array([[cos(ang[1]),0,sin(ang[1])],[0,1,0],[-np.sin(ang[1]),0,cos(ang[1])]])@acc_stride_plus[kk,:].T).T-np.array([9.81,0,0]) for kk,ang in enumerate(theta)]
```
Comment (EricHonertBOA) [inline Exploration_SampleEntropy_IMU.py:279]: I double checked the rotation matrix (fully written out from 288-303). They are consistent with the matrix multiplication that we are doing.

**Review by EricHonertBOA — COMMENTED** [review]
I double checked the rotation matrices - they are correct. The problem you are encountering is probably due to mis-assigning the accelerometer and gyroscope signals in your function

**Comment by kharrison89** [comment]
Hey hey, I think this pull request fell off our radar. If you have a sec to check it out and approve, Bethany needs updated code for skiing.

**Review by EricHonertBOA — APPROVED** [review]

**Pull request merged**